### PR TITLE
[Coupons] Product selector enhancements

### PR DESCRIPTION
**Pull request by 0nko**
This PR adds a couple of minor enhancements/features to the product selector:

- Added the Done button, which returns selected products back to the edit coupon screen
- Added the Clear button to clear the selection
- Selected products state preservation during process death
- Fixed various naming and styling issues (button titles, paddings)
- Partial selection checkmark for variable products

Before merging this, please merge #6571 first.

https://user-images.githubusercontent.com/1522856/170484752-63111265-ce9d-4dd6-bf09-3ee3529d0f39.mp4

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/selector/ProductSelectorViewModel.kt [hunk 1]
```diff
@@ -84,10 +88,25 @@ class ProductSelectorViewModel @Inject constructor(
...
+        fun getProductSelection(): SelectionState {
+            return if (productType == VARIABLE && numVariations > 0) {
+                val intersection = variationIds.intersect(selectedIds.toSet())
+                if (intersection.isEmpty()) {
+                    UNSELECTED
+                } else if (intersection.size < variationIds.size) {
+                    PARTIALLY_SELECTED
+                } else {
+                    return SELECTED
+                }
```
Comment (hafizrahman) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/selector/ProductSelectorViewModel.kt]: np/question: AS suggested this to be a when instead, so something like:

```
                when {
                    intersection.isEmpty() -> UNSELECTED
                    intersection.size < variationIds.size -> PARTIALLY_SELECTED
                    else -> SELECTED
                }
```

I'm curious if you decided to use multiple if's for better readability though, in which case I'd like to learn more about it.

**Review by hafizrahman — APPROVED** [review]
Looks great to me! Just left a question but otherwise everything seems to work and behave properly.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/selector/ProductSelectorScreen.kt [hunk 2]
```diff
@@ -161,20 +192,27 @@ private fun ProductList(
...
+private fun getDoneButtonTitle(selectedItemsCount: Int) =
+    if (selectedItemsCount > 0) {
+        StringUtils.getQuantityString(
+            quantity = selectedItemsCount,
+            default = R.string.product_selector_select_button_title_default,
+            one = R.string.product_selector_select_button_title_one
+        )
+    } else {
+        stringResource(R.string.done)
+    }
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/selector/ProductSelectorScreen.kt]: np, we can simplify this by passing the `zero` argument with the String `stringResource(R.string.done)`, we can then get rid of the `if/else`

File: .idea/inspectionProfiles/Project_Default.xml [hunk 3]
```diff
@@ -11,6 +11,24 @@
...
+    </inspection_tool>
+    <inspection_tool class="PreviewMustBeTopLevelFunction" enabled="true" level="ERROR" enabled_by_default="true">
+      <option name="previewFile" value="true" />
+    </inspection_tool>
+    <inspection_tool class="PreviewNeedsComposableAnnotation" enabled="true" level="ERROR" enabled_by_default="true">
+      <option name="previewFile" value="true" />
+    </inspection_tool>
+    <inspection_tool class="PreviewPickerAnnotation" enabled="true" level="ERROR" enabled_by_default="true">
+      <option name="previewFile" value="true" />
+    </inspection_tool>
```
Comment (hichamboushaba) [inline .idea/inspectionProfiles/Project_Default.xml]: I think this is an accidental commit, can you revert it?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/selector/ProductSelectorScreen.kt (hunk 2 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/selector/ProductSelectorScreen.kt]: That's true, thanks!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/selector/ProductSelectorViewModel.kt (hunk 1 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/selector/ProductSelectorViewModel.kt]: No. I usually go with the suggestion but I didn't notice it here. I'll update it.

**Comment by 0nko** [comment]
@hafizrahman @hichamboushaba Thank you both, I've addressed your comments and the PR's ready for another round.

**Review by hichamboushaba — APPROVED** [review]
Looks good, thanks for the changes @0nko

**Pull request merged**